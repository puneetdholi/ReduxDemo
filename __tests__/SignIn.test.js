import React from 'react';
import renderer from 'react-test-renderer';
import SignIn from '../src/screens/SignIn';
import {TextInput} from 'react-native';
import {render, fireEvent, act} from '@testing-library/react-native';

test('test renders correctly', () => {
  const tree = renderer.create(<SignIn />).toJSON;
  expect(tree).toMatchSnapshot();
});

it('renders default elements', () => {
  const {getAllByText, getByPlaceholderText} = render(<SignIn />);

  expect(getAllByText('Login').length).toBe(2);
  getByPlaceholderText('example');
  getByPlaceholderText('***');
});

it('shows invalid text input messages', () => {
  const {getByTestId, getByText} = render(<SignIn />);

  fireEvent.press(getByTestId('SignIn.Button'));

  getByText('Invalid username.');
  getByText('Invalid password.');
});
