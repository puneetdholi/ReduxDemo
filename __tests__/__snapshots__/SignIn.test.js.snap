// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test renders correctly 1`] = `[Function]`;
